components/board/genericlinux/gpio.go

```diff
@@ -276,15 +290,19 @@ func (pin *gpioPin) halfPwmCycle(shouldBeOn bool) bool {
 		dutyCycle = 1 - dutyCycle
 	}
 	duration := time.Duration(float64(time.Second) * dutyCycle / float64(freqHz))
-	return accurateSleep(pin.cancelCtx, duration)
+
+	return accurateSleep(ctx, duration)
 }
 
-func (pin *gpioPin) softwarePwmLoop() {
+func (pin *gpioPin) softwarePwmLoop(ctx context.Context) {
 	for {
-		if !pin.halfPwmCycle(true) {
+		if err := ctx.Err(); err != nil {
+			return
+		}
```

Comment:
My feedback on line 272 seems redundant with this new if statement. It probably can't hurt to have both, but it's not clear to me that we need this one at all.

Reply:
Yeah was also unsure if we needed both, I will remove this one.